### PR TITLE
Run all tests when apis change

### PR DESCRIPTION
**Pull request by advayDev1**

### CODE REVIEW

File: .github/workflows/flutter-build.yml [hunk 1]
```diff
@@ -3,11 +3,13 @@ on:
   push:
     paths:
       - '.github/workflows/**'
+      - 'api/**'
```
Comment (hspinks) [inline .github/workflows/flutter-build.yml:6]: will the Flutter build depend on API changes?

File: .github/workflows/flutter-test.yml [hunk 2]
```diff
@@ -3,11 +3,13 @@ on:
   push:
     paths:
       - '.github/workflows/**'
+      - 'api/**'
```
Comment (hspinks) [inline .github/workflows/flutter-test.yml:6]: Same question... if our Flutter tests are unit tests and not integration tests, they wouldn't depend on API changes. I think there's a chance of false positive tests that give us a false sense of security (e.g. the API could change, we don't update our Flutter tests to the new API spec, Flutter unit tests still pass, but app runs fail).

Unless we think that we'll be writing Flutter tests that actually hit the APIs, but that would be a different conversation...

File: .github/workflows/flutter-build.yml (hunk 1 above)
Comment (advayDev1) [reply .github/workflows/flutter-build.yml:6]: The API directory defines both the service and also the data structures.  The Flutter client hand-rolls the JSON but it should use for example protoc-generated dart classes: https://developers.google.com/protocol-buffers/docs/darttutorial

File: .github/workflows/flutter-test.yml (hunk 2 above)
Comment (advayDev1) [reply .github/workflows/flutter-test.yml:6]: Its not about hitting the APIs.  Its that the client should compile based on protoc-generated dart classes from the protobufs in api/

**Pull request merged**